external/fv3fit/fv3fit/reservoir/autoencoder.py

```diff
@@ -0,0 +1,43 @@
+import os
+import tensorflow as tf
+from fv3fit._shared import get_dir, put_dir
+
+
+class Autoencoder(tf.keras.Model):
```

Comment:
Please add type hints to the methods of this function. Some docstrings would also be nice, but type hints are good enough.

For the models, you could define a protocol class:
```from typing import Protocol

class Model(Protocol):

    def predict(self, x: tf.Tensor) -> tf.Tensor:
        ...

    def __call__(self, x: tf.Tensor) -> tf.Tensor:
        ...

    def save(self, path: str) -> None:
        ...
```

Reply:
Scratch that, just use tf.keras.Model for models. I just realized dump and then load doesn't work if it's not a keras model.